### PR TITLE
Adding doorkeeper into the stack

### PR DESCRIPTION
**Pull request by vpetrusenko**

### CODE REVIEW

**Comment by mod** [comment]
Looks ok @gfedorenko we will need to restest that branch.
Does it works with peatio ?
We need a client application for testing.

File: db/migrate/20180109194005_devise_create_accounts.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class DeviseCreateAccounts < ActiveRecord::Migration[5.1]
+class DeviseCreateAccounts < ActiveRecord::Migration[5.0] # because with 5.1 Column `application_id` has a type of `bigint(20)`
```
Comment (gfedorenko) [inline db/migrate/20180109194005_devise_create_accounts.rb]: column account_id

**Comment by vpetrusenko** [comment]
https://github.com/vpetrusenko/barongClient  client for testing

File: Gemfile [hunk 2]
```diff
@@ -13,6 +13,9 @@ gem 'cancancan',    '~> 2.0'
 gem 'bootstrap',    '~> 4.0.0.beta3'
 gem 'jquery-rails', '~> 4.3'
 gem 'kaminari',     '~> 1.1.1'
+gem 'doorkeeper'
```
Comment (gfedorenko) [inline Gemfile]: Define the version of gems

File: app/views/doorkeeper/applications/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      <%= doorkeeper_errors_for application, :scopes %>
+      <span class="help-block">
+        <%= t('doorkeeper.applications.help.scopes') %>
+      </span>
+    </div>
+  <% end %>
+
+  <div class="form-group">
+    <div class="col-sm-offset-2 col-sm-10">
+      <%= f.submit t('doorkeeper.applications.buttons.submit'), class: "btn btn-primary" %>
```
Comment (gfedorenko) [inline app/views/doorkeeper/applications/_form.html.erb]: single quotes for class

File: app/views/doorkeeper/applications/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      <%= link_to t('doorkeeper.applications.buttons.cancel'), oauth_applications_path, :class => "btn btn-default" %>
```
Comment (gfedorenko) [inline app/views/doorkeeper/applications/_form.html.erb]: rewrite as class: 'btn btn-default'

File: app/views/doorkeeper/authorizations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  <% end %>
+
+  <div class="actions">
+    <%= form_tag oauth_authorization_path, method: :post do %>
+      <%= hidden_field_tag :client_id, @pre_auth.client.uid %>
+      <%= hidden_field_tag :redirect_uri, @pre_auth.redirect_uri %>
+      <%= hidden_field_tag :state, @pre_auth.state %>
+      <%= hidden_field_tag :response_type, @pre_auth.response_type %>
+      <%= hidden_field_tag :scope, @pre_auth.scope %>
+      <%= submit_tag t('doorkeeper.authorizations.buttons.authorize'), class: "btn btn-success btn-lg btn-block" %>
```
Comment (gfedorenko) [inline app/views/doorkeeper/authorizations/new.html.erb]: single quotes for class

File: app/views/doorkeeper/authorizations/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    <% end %>
+    <%= form_tag oauth_authorization_path, method: :delete do %>
+      <%= hidden_field_tag :client_id, @pre_auth.client.uid %>
+      <%= hidden_field_tag :redirect_uri, @pre_auth.redirect_uri %>
+      <%= hidden_field_tag :state, @pre_auth.state %>
+      <%= hidden_field_tag :response_type, @pre_auth.response_type %>
+      <%= hidden_field_tag :scope, @pre_auth.scope %>
+      <%= submit_tag t('doorkeeper.authorizations.buttons.deny'), class: "btn btn-danger btn-lg btn-block" %>
```
Comment (gfedorenko) [inline app/views/doorkeeper/authorizations/new.html.erb]: single quotes for class

File: app/views/layouts/doorkeeper/admin.html.erb [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html>
+<head>
```
Comment (gfedorenko) [inline app/views/layouts/doorkeeper/admin.html.erb]: fix indentation

File: app/views/layouts/doorkeeper/application.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html>
+<head>
```
Comment (gfedorenko) [inline app/views/layouts/doorkeeper/application.html.erb]: fix indentation

File: Gemfile [hunk 9]
```diff
@@ -13,6 +13,9 @@ gem 'cancancan',    '~> 2.0'
 gem 'bootstrap',    '~> 4.0.0.beta3'
 gem 'jquery-rails', '~> 4.3'
 gem 'kaminari',     '~> 1.1.1'
+gem 'doorkeeper',   '~> 4.2.6'
```
Comment (mod) [inline Gemfile]: why there is ominauth ?

**Pull request merged**